app/agora.py

```diff
@@ -76,24 +81,27 @@ def build_node(node, extension='', user_list=''):
     node = urllib.parse.unquote_plus(node)
     node = util.slugify(node)
 
-    n = copy(G.node(node))
+    # n = copy(G.node(node))
+    n = db.query_node(node)
```

Comment:
IIUC this PR adds a hard dependency on the .sqlite file for the Agora to work. I don't think we want to do that. The Agora should continue to work (albeit slowly/with a reduced feature set) without .sqlite -- at least while we're experimenting. Then we can have a checkpoint to make the .sqlite file a hard dependency. Until then, I think we should use sqlite where available but keep code paths free of sqlite and working. Ideally we should also have an ENABLE_SQLITE in config.py to make this easier to manage. Wdyt?

Reply:
I keep vacillating on this. if its optional I don't think we should even add sqlite. just needless complexity without measurable increase.

Reply:
I keep vacillating on this. if its optional I don't think we should even add sqlite. just needless complexity without measurable increase.

Reply:
I know what you mean! But I keep wondering: why can't it be like Javascript? The Agora runs without Javascript; it's just better (more featureful) when you enable it. We could take the same approach with sqlite/other databases? The Agora works on the filesystem only, which makes it so that starting an Agora is just launching the server and making sure the right files are in the right paths; but if you run an extra process that generates sqlite, you get goodies.

Reply:
>I know what you mean! But I keep wondering: why can't it be like Javascript? The Agora runs without Javascript; The Agora runs without Javascript; it's just better (more featureful) when you enable it.

I kinda feel like this is saying "computers run without an operating system, they're just more featureful when you have one"